copy/copy.go

```diff
@@ -1098,6 +1098,18 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 					Artifact: srcInfo,
 				}
 			}
+
+			// If we know the blob that we just reused is compressed (or not), note how it's compressed (or not).
```

Comment:
Do I understand it correctly that this does the same thing the `OperationAndAlgorithmForCompressor` call + use of its return value does in `dockerImageDest.TryReusingBlob` — and essentially overrides that — basically because the blob cache doesn’t / currently can’t use BlobInfoCache2?

If so, this should probably use `OperationAndAlgorithmForCompressor` and update _both_ fields, so that it does truly overwrite the update done by `TryReusingBlob`.

Reply:
When the blob cache is wrapping a destination, it ends up calling the destination's `TryReusingBlob` or `PutBlob`, so it's not in control of what's in the returned `BlobInfo`.  The code path where we hit this is when `TryReusingBlob` finds the target blob is already in the repository (around [here](https://github.com/containers/image/blob/1da72bf69b0f111b898fe7d87bf652d634dcd053/docker/docker_image_dest.go#L305)) but doesn't supply the info from the cache.
I _think_ that updating `TryReusingBlob` to always fill in this information on this path would also work, but there are multiple `TryReusingBlob` implementations, so having the copy package supply the information looked like it'd require fewer changes.

Reply:
> When the blob cache is wrapping a destination, it ends up calling the destination's `TryReusingBlob` or `PutBlob`, so it's not in control of what's in the returned `BlobInfo`. 

Just to confirm: `blobCacheDestination.PutBlob` never substitutes with a different digest, so in that case all the correct info should have already been in the source manifest and/or `LayerInfosForCopy`; is that correct?